rest_docs/rest-catalog-open-api.yaml

```diff
@@ -0,0 +1,712 @@
+        required: true
+      responses:
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/responses/UpdatePropertiesResponse'
+              example: {
+                "updated": [ "owner" ],
+                "removed": [ "foo" ],
+                "missing": [ "bar" ]
+              }
+        404:
+          description: Not Found
+        406:
+          description: Not Acceptable - Unsupported Operation
+        422:
+          description: Unprocessable Entity - A property key was included in both removables and updates
+
+  /v1/namespaces/{namespace}/tables:
+    parameters:
+      - $ref: '#/components/parameters/namespace'
+    get:
+      tags:
+        - Catalog API
+      summary: List all table identifiers underneath a given namespace
+      description: Return all table identifiers under this namespace
+      operationId: listTables
+      responses:
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  identifiers:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/TableIdentifier'
+              examples:
+                TableIdentifierListExample:
+                  $ref: '#/components/examples/ListTablesExample'
+        404:
+          description: Not Found - Either the namespace or the table does not exist
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ResponseErrorObject'
+              examples:
+                NamespaceNotFound:
+                  $ref: '#/components/examples/NoSuchNamespaceError'
+                TableNotFound:
+                  $ref: '#/components/examples/NoSuchTableError'
+
+  /v1/namespaces/{namespace}/tables/{table}:
+    parameters:
+      - $ref: '#/components/parameters/namespace'
+      - $ref: '#/components/parameters/table'
+    delete:
+      tags:
+        - Catalog API
+      summary: Drop a table from the catalog
+      operationId: dropTable
+      description: Remove a table from the catalog
+      parameters:
+        - name: purgeRequested
+          in: query
+          required: false
+          description: Whether the user requested to purge the underlying table's data and metadata
+          schema:
+            type: boolean
+            default: false
+      responses:
+        200:
+          description: OK
+          content:
+            'application/json':
+              schema:
+                $ref: '#/components/responses/DropTableResponse'
+              example: { "dropped": true, "purged": false }
+        202:
+          description: Accepted - for use if purgeRequested is implemented as an asynchronous action.
+        404:
+          $ref: '#/components/schemas/ResponseErrorObject'
```

Comment:
What's the difference between this and the `content` metadata like you use for `rename`? Can we make these consistent?

Reply:
This is simply less overhead than having to repeat all of the layers for `content: application/json: schema` for every single 404 route.

I can do it the way that that `rename` is, but it will make the document a lot larger.

Reply:
Let me know if you prefer the `content` metadata or this shorthand.

Reply:
The difference would be 8 additional lines per every single response code (202, 404, 406, 422) vs zero or one additional line.

But the content way is probably more standard (but will make the document a lot larger).

Reply:
Found a way to do it without making the document _too much_ larger. There are some quirks to working with an Open API doc I didn't know before so settling on what should be standardized will be helpful.